frontend/public/components/monitoring/query-browser.tsx

```diff
@@ -731,10 +740,13 @@ const QueryBrowser_: React.FC<QueryBrowserProps> = ({
     );
   }
 
-  const onZoom = (from: number, to: number) => {
+  const zoomableGraphOnZoom = (from: number, to: number) => {
     setXDomain([from, to]);
     setSpan(to - from);
     setSamples(defaultSamples || getMaxSamplesForSpan(to - from));
+    if (_.isFunction(onZoom)) {
+      onZoom(from, to);
+    }
```

Comment:
Maybe just use optional chaining. Presumably our types will ensure it's a function if it's defined.

```suggestion
    onZoom?.(from, to);
```

Reply:
👍